lib/transition/import/csv/import_batch_row.rb

```diff
@@ -60,6 +64,8 @@ def <=>(other)
             1
           elsif archive? && other.redirect?
             -1
+          elsif archive? && other.archive?
+            other.line_number <=> line_number
```

Comment:
I'm not sure whether this behaviour is the right thing to do, because it would be difficult for users to know how to reliably get their custom archive URLs saved.

For example, say a user had noticed that `http://foo.gov.uk/a?x=y` needed a custom archive URL and imported a CSV including these two rows (in this order):

```
http://foo.gov.uk/a?b=c,TNA
http://foo.gov.uk/a?x=y,http://webarchive.nationalarchives.gov.uk/20120816224015/http://foo.gov.uk/a
```

Those rows would both have the same canonical path `/a`, and with the current behaviour the earlier one would take precedence, losing the custom archive URL.

To get around this, the user would have to know that they need to add the custom archive URL to the earliest row in the CSV which would canonicalize to that path in order for it to be saved.

I think it would appear more consistent to users for an archive row with a custom archive URL to take precedence over earlier normal archive rows. If there are multiple rows with custom archive URLs then we should fall back to using the earliest one. This would increase the complexity of the code, but I think it's probably worth it for the more understandable behaviour for users.